docs/userguide.md

```diff
@@ -18,42 +27,48 @@ export PATH=$(pwd)/bin:${PATH}
 
 Or you can install them manually yourself using the guidelines provided below.
 
-#### apiserver-builder
+#### kubebuilder
 
 This repo depends on
-[apiserver-builder](https://github.com/kubernetes-incubator/apiserver-builder)
-to generate code and build binaries. Download a [recent
-release](https://github.com/kubernetes-incubator/apiserver-builder/releases)
-and install it in your `PATH`.
-
-#### cluster-registry
-
-Get version 0.0.4 of the [Cluster
-Registry](https://github.com/kubernetes/cluster-registry/releases/tag/v0.0.4)
+[kubebuilder](https://github.com/kubernetes-sigs/kubebuilder)
+to generate code and build binaries. Download the [v0.1.12
+release](https://github.com/kubernetes-sigs/kubebuilder/releases/tag/v0.1.12)
 and install it in your `PATH`.
 
 #### kubectl
 
 Download a version of
-[`kubectl`](https://kubernetes.io/docs/tasks/tools/install-kubectl/) and
+[`kubectl`](https://kubernetes.io/docs/tasks/tools/install-kubectl/) >= v1.11 and
 install it in your `PATH`.
 
+#### kube-apiserver
+
+Download a version of
+[`kube-apiserver`](https://github.com/kubernetes/kubernetes/releases) >= 1.11
+and install it in your `PATH`. Make sure the path to this binary comes before
+the path to the `kubebuilder` installation, otherwise you will use the
+`kube-apiserver` from `kubebuilder` instead. To download version 1.11:
+   - https://storage.googleapis.com/kubernetes-release/release/v1.11.0/bin/linux/amd64/kube-apiserver
+
 ### Create Clusters
 
 The quickest way to set up clusters for use with the federation-v2 control
 plane is to use [minikube](https://kubernetes.io/docs/getting-started-guides/minikube/).
 
-**NOTE: You will need to use minikube version
-[0.25.2](https://github.com/kubernetes/minikube/releases/tag/v0.25.2) in order
-to avoid an issue with profiles, which is required for testing multiple
-clusters. See issue https://github.com/kubernetes/minikube/issues/2717 for more
-details.**
+**NOTE: You will need to use a minikube version that supports deploying a
+kubernetes cluster >= 1.11. Currently there is no released version of minikube
+that supports kube v1.11 with profiles so you'll need to either:**
+
+1. Build minikube from master by following these
+   [instructions](https://github.com/kubernetes/minikube/blob/master/docs/contributors/build_guide.md).
+2. Or use a recent CI build such as [this one from PR
+   2943](http://storage.googleapis.com/minikube-builds/2943/minikube-linux-amd64).
 
 Once you have minikube installed run:
 
 ```bash
-minikube start -p cluster1
-minikube start -p cluster2
+minikube start -p cluster1 --kubernetes-version v1.11.0
```

Comment:
Does this mean that federation v2 do not support managing cluster pre v1.11.0 release?

Reply:
In theory, it can manage pre v1.11.0 clusters for the other clusters that are joined to the federation, but the host cluster must be v1.11.0. We do not have a version compatibility matrix and we need to discuss our overall version support strategy.

Reply:
Yes, so for now, how about highlight this in the document? Such as we need make sure the host cluster must be v1.11.0 but the joining clusters can be other versions?

Reply:
The doc explicitly describes the need for k8s version >= 1.11. I do not want to add any comments about using older versions of k8s in the joining clusters because we do not necessarily support that and have not tested it. In theory it may work, but we do not have a strategy for version compatibility yet. I think we should expect all clusters to be >= 1.11 at least for the first alpha release.

Reply:
@font then this will be a big limitation for federation v2. Kubernetes 1.11 is the latest version and many customers are still using kubenretes 1.10, 1.9, 1,8 etc and most of them actually want manage those versions.

I have did some test with 1.11 host cluster, and 1.10 as join cluster, it works fine, so can we update the note a bit to clarify we only need host cluster to be 1.11, but the joining cluster does not request 1.11 and it can be a lower version?

Reply:
Also append my test result here.

```
root@gyliu-dev1:~/go/src/github.com/kubernetes-sigs/federation-v2# kubectl get nodes
NAME             STATUS    ROLES     AGE       VERSION
172.16.250.138   Ready     <none>    5d        v1.11.0
172.16.250.140   Ready     <none>    5d        v1.11.0
```
```
root@gyliu-dev1:~/go/src/github.com/kubernetes-sigs/federation-v2# kubectl get nodes --context cluster141.k8s.local
NAME             STATUS    ROLES     AGE       VERSION
172.16.250.141   Ready     <none>    22m       v1.10.0+icp-ee
172.16.250.142   Ready     <none>    21m       v1.10.0+icp-ee
```
```
root@gyliu-dev1:~/go/src/github.com/kubernetes-sigs/federation-v2# for r in configmaps secrets deploy; do     for c in cluster138.k8s.local cluster141.k8s.local; do         echo; echo ------------ ${c} ------------; echo;         kubectl --context=${c} -n test-namespace get ${r};         echo; echo;     done; done

------------ cluster138.k8s.local ------------

NAME             DATA      AGE
test-configmap   1         4m



------------ cluster141.k8s.local ------------

NAME             DATA      AGE
test-configmap   1         4m



------------ cluster138.k8s.local ------------

NAME                  TYPE                                  DATA      AGE
default-token-bb5kr   kubernetes.io/service-account-token   3         5m
test-secret           Opaque                                1         4m



------------ cluster141.k8s.local ------------

NAME                  TYPE                                  DATA      AGE
default-token-tg5cp   kubernetes.io/service-account-token   3         4m
test-secret           Opaque                                1         4m



------------ cluster138.k8s.local ------------

NAME              DESIRED   CURRENT   UP-TO-DATE   AVAILABLE   AGE
test-deployment   3         3         3            3           4m



------------ cluster141.k8s.local ------------

NAME              DESIRED   CURRENT   UP-TO-DATE   AVAILABLE   AGE
test-deployment   5         5         5            5           4m
```

Reply:
@gyliu513 We need to discuss version compatibility support more broadly with the entire group before we decide to put any notes around what versions we support.